backends/cassandra.go

```diff
@@ -5,50 +5,94 @@ import (
 
 	"github.com/gocql/gocql"
 	"github.com/prebid/prebid-cache/config"
+	"github.com/prebid/prebid-cache/utils"
 	log "github.com/sirupsen/logrus"
 )
 
-// Cassandra Object use to implement backend interface
-type Cassandra struct {
+type CassandraDB interface {
+	Get(ctx context.Context, key string) (string, error)
+	Put(ctx context.Context, key string, value string, ttlSeconds int) error
+	Init() error
+}
+
+// CassandraDBClient is a wrapper for the Cassandra client that
+// interacts with the Cassandra server and implements the CassandraDB client
+type CassandraDBClient struct {
+	cfg     config.Cassandra
 	cluster *gocql.ClusterConfig
 	session *gocql.Session
 }
 
-// NewCassandraBackend create a new cassandra backend
-func NewCassandraBackend(cfg config.Cassandra) *Cassandra {
-	var err error
+func (c *CassandraDBClient) Get(ctx context.Context, key string) (string, error) {
+	var res string
+
+	err := c.session.Query(`SELECT value FROM cache WHERE key = ? LIMIT 1`, key).
+		WithContext(ctx).
+		Consistency(gocql.One).
+		Scan(&res)
 
-	c := &Cassandra{}
-	c.cluster = gocql.NewCluster(cfg.Hosts)
-	c.cluster.Keyspace = cfg.Keyspace
+	return res, err
+}
+
+func (c *CassandraDBClient) Put(ctx context.Context, key string, value string, ttlSeconds int) error {
+	var insertedKey, insertedValue string
+	success, err := c.session.Query(`INSERT INTO cache (key, value) VALUES (?, ?) IF NOT EXISTS USING TTL ?`, key, value, ttlSeconds).
+		WithContext(ctx).
+		ScanCAS(&insertedKey, &insertedValue)
+
+	if !success {
+		return utils.RecordExistsError{}
+	}
+	return err
+}
+
+// Init initializes cassandra cluster and session with the configuration
+// loaded from environment variables or configuration files at startup
+func (c *CassandraDBClient) Init() error {
+	c.cluster = gocql.NewCluster(c.cfg.Hosts)
+	c.cluster.Keyspace = c.cfg.Keyspace
 	c.cluster.Consistency = gocql.LocalOne
 
+	var err error
 	c.session, err = c.cluster.CreateSession()
-	if err != nil {
+
+	return err
+}
+
+//------------------------------------------------------------------------
```

Comment:
Nitpick: Please remove the line comment.

Reply:
I kind of dig the lines in order to separate the client from the backend TBH. Removed anyways.